### PR TITLE
Dependency overhaul

### PR DESCRIPTION
**Pull request by TrueSkrillor**
**DO NOT MERGE until PRs for dependant projects are available**

tl;dr:

- Update log4j to version 2.17.2
- Complete JUnit 5 migration
- Replaced commons-lang with commons-text
- Replaced javax.xml.bind with jakarta.xml.bind
- Restructured pom.xml
- Bumped maven plugins to latest versions

### JUnit 5 Migration

This PR migrates all unit tests to JUnit 5 without adding or removing test cases. As a result, the JUnit 4 dependency has been removed entirely. Empty placeholder test methods have been disabled via the new `@Disabled` annotation to speed up test execution and signal a skipped test rather than a successful one.

### `commons-lang` replaced by `commons-text`

The project only depends on the StringEscapeUtils class from commons-lang. With commons-lang3, this class was moved to the commons-text library instead. Therefore, we can safely substitute commons-lang with the newer commons-text.

### `javax.xml.bind` replaced by `jakarta.xml.bind`

The `javax.xml.bind` Java EE module was removed by Oracle with the release of Java 11. Although the deprecated module is available within the maven repository for backwards compatibility, one should replace all usages with the `jakarta.xml.bind` library. The `jakarta.xml.bind` is a drop-in replacement without the need to change any code other than the imports.

### CODE REVIEW

**Review by NErinola — COMMENTED** [review]
I reviewed the changes but did not test anything yet. We will do so once the other PRs are reviewed so we can build and test everything in one go.

**Comment by TrueSkrillor** [comment]
I bumped the Log4J and plugin versions one last time before merge. A rebase is not necessary for this PR. This PR is now ready to be merged into master.

**Review by mmaehren — APPROVED** [review]
